lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/fs.rb

```diff
@@ -976,6 +978,7 @@ def cmd_upload(*args)
 
     # Go through each source item and upload them
     src_items.each { |src|
+      src = ::File.expand_path(src)
```

Comment:
If we are expanding the source then wouldn't it also be reasonable to expand the destination as well here?

Reply:
See https://github.com/rapid7/metasploit-framework/pull/13954#discussion_r518360486, r.e my comment on this; at the moment looks like this might not be possible due to a bug. It might be possible to fix this with `cmd_exec` as a patch until Meterpreter can be fixed, but I'm not sure.